exporter/elasticsearchexporter/internal/serializer/otelserializer/serializeprofiles/transform.go

```diff
@@ -0,0 +1,416 @@
+	return stackPayload, nil
+}
+
+// symbolizedFrames returns a slice of StackFrames that have symbols.
+func symbolizedFrames(frames []StackFrame) []StackFrame {
+	framesWithSymbols := make([]StackFrame, 0, len(frames))
+	for i := range frames {
+		if isFrameSymbolized(frames[i]) {
+			framesWithSymbols = append(framesWithSymbols, frames[i])
+		}
+	}
+	return framesWithSymbols
+}
+
+func isFrameSymbolized(frame StackFrame) bool {
+	return len(frame.FileName) > 0 || len(frame.FunctionName) > 0
+}
+
+func stackTraceEvent(traceID string, profile pprofile.Profile, sample pprofile.Sample) StackTraceEvent {
+	event := StackTraceEvent{
+		StackTraceID: traceID,
+		Count:        1, // TODO: Check whether count can be dropped with nanosecond timestamps
+	}
+
+	// Store event-specific attributes.
+	for i := 0; i < sample.AttributeIndices().Len(); i++ {
+		if profile.AttributeTable().Len() < i {
+			continue
+		}
+		attr := profile.AttributeTable().At(i)
+
+		switch attribute.Key(attr.Key()) {
+		case semconv.ContainerIDKey:
+			event.ContainerID = attr.Value().AsString()
+		case semconv.K8SPodNameKey:
+			event.PodName = attr.Value().AsString()
+		case semconv.ContainerNameKey:
+			event.ContainerName = attr.Value().AsString()
+		case semconv.ThreadNameKey:
+			event.ThreadName = attr.Value().AsString()
+		}
+	}
+
+	return event
+}
+
+func stackTrace(stackTraceID string, frames []StackFrame, frameTypes []libpf.FrameType) StackTrace {
+	frameIDs := make([]string, 0, len(frames))
+	for _, f := range frames {
+		frameIDs = append(frameIDs, f.DocID)
+	}
+
+	// Up to 255 consecutive identical frame types are converted into 2 bytes (binary).
+	// We expect mostly consecutive frame types in a trace. Even if the encoding
+	// takes more than 32 bytes in single cases, the probability that the average base64 length
+	// per trace is below 32 bytes is very high.
+	// We expect resizing of buf to happen very rarely.
+	buf := bytes.NewBuffer(make([]byte, 0, 32))
+	encodeFrameTypesTo(buf, frameTypes)
+
+	return StackTrace{
+		DocID:    stackTraceID,
+		FrameIDs: strings.Join(frameIDs, ""),
+		Types:    buf.String(),
+	}
+}
+
+func stackFrames(profile pprofile.Profile, sample pprofile.Sample) ([]StackFrame, []libpf.FrameType, *libpf.FrameID, error) {
+	frames := make([]StackFrame, 0, sample.LocationsLength())
+
+	locations := getLocations(profile, sample)
+	totalFrames := 0
+	for _, location := range locations {
+		totalFrames += location.Line().Len()
+	}
+	frameTypes := make([]libpf.FrameType, 0, totalFrames)
+
+	var leafFrameID *libpf.FrameID
+
+	for locationIdx, location := range locations {
+		if location.MappingIndex() >= int32(profile.MappingTable().Len()) {
+			continue
+		}
+
+		frameTypeStr, err := getStringFromAttribute(profile, location, "profile.frame.type")
+		if err != nil {
+			// Fallback option if the frame type is still provided via the
+			// deprecated Location.type_index field.
+			frameTypeStr = profile.StringTable().At(int(location.MappingIndex()))
```

Comment:
iirc `Location.type_index` got dropped before the OTel profiling protocol got moved to `v1development`. As OTel collector components build on `v1development`, I think, legacy burden should be dropped.
```suggestion
			return nil, nil, nil, err
```